js/researches/getPassiveVoice.js

```diff
@@ -37,12 +54,14 @@ var matchArray = function( sentence, matches ) {
 
 	forEach( matches, function( part ) {
 		part = stripSpaces( part );
+		var indices = getIndicesOf( part, sentence );
 
 		if ( !matchWordInSentence( part, sentence ) ) {
 			return;
 		}
-
-		matchedParts.push( { index: sentence.indexOf( part ), match: part } );
+		forEach( indices, function( index ) {
+			matchedParts.push( { index: index, match: part } );
```

Comment:
Aren't you more or less doing the same thing here as you did in `getIndicesOf()`? Wouldn't it make more sense to do all these actions in a single loop?